packages/frontend-v2/validation-worker/worker.ts

```diff
@@ -0,0 +1,16 @@
+import { validateMajor2 } from "@graduate/common";
+import { Loaded, ValidationResult, WorkerMessageType, WorkerPostInfo } from "./worker-messages";
+
+// Let the host page know the worker is ready.
+const loadMessage: Loaded = {type: WorkerMessageType.Loaded}
+postMessage(loadMessage)
+
+addEventListener("message", ({data}: MessageEvent<WorkerPostInfo>) => {
+  const validationResult: ValidationResult = {
+    type: WorkerMessageType.ValidationResult,
+    result: validateMajor2(data.major, data.taken, data.concentration),
+    requestNumber: data.requestNumber
+  }
+  
+  postMessage(validationResult)
+})
```

Comment:
Without the types and race condition logic, the actual worker script is very simple now :)